pkg/clients/database/sql.go

```diff
@@ -0,0 +1,612 @@
+	s := cr.Spec.ForProvider
+	properties := &mysql.ServerPropertiesForDefaultCreate{
+		AdministratorLogin:         azure.ToStringPtr(s.AdministratorLogin),
+		AdministratorLoginPassword: &adminPassword,
+		Version:                    mysql.ServerVersion(s.Version),
+		SslEnforcement:             mysql.SslEnforcementEnum(s.SSLEnforcement),
+		CreateMode:                 mysql.CreateModeDefault,
+		StorageProfile: &mysql.StorageProfile{
+			BackupRetentionDays: azure.ToInt32PtrFromIntPtr(s.StorageProfile.BackupRetentionDays),
+			GeoRedundantBackup:  mysql.GeoRedundantBackup(azure.ToString(s.StorageProfile.GeoRedundantBackup)),
+			StorageMB:           azure.ToInt32Ptr(s.StorageProfile.StorageMB),
+			StorageAutogrow:     mysql.StorageAutogrow(azure.ToString(s.StorageProfile.StorageAutogrow)),
+		},
+	}
+	sku, err := ToMySQLSKU(s.SKU)
+	if err != nil {
+		return err
+	}
+	createParams := mysql.ServerForCreate{
+		Sku:        sku,
+		Properties: properties,
+		Location:   &s.Location,
+		Tags:       azure.ToStringPtrMap(s.Tags),
+	}
+	op, err := c.Create(ctx, s.ResourceGroupName, meta.GetExternalName(cr), createParams)
+	if err != nil {
+		return err
+	}
+	// NOTE(muvaf): There are cases where Create call does not return error
+	// and you don't see any SQL instance in the console UI. In those cases,
+	// since name is not taken, we always call Create with a new password each time.
+	// The problem is, error that blocks creation happens after the Create call
+	// is initiated.
+	// DoneWithContext checks the operation once and errors usually surface
+	// themselves in the first check, if there is one we are able to warn the user
+	// for a fix.
+	// However, there could be cases where the error appears after a long while.
+	// For that reason, some libraries make use of WithCompletionRef function. However,
+	// that function blocks until success or error with the context deadline.
+	// Azure does not provide any guarantees regarding provisioning time and it
+	// can get really long. In most of the cases, our context deadline gets reached.
+	//
+	// For the time being, we check only once to cover the greater possibility of
+	// capturing the error. If there is none, we'll never call Create again anyway.
+	// If the problem happens after a while, we'd fallback to old behavior where
+	// we continuously try to create and fail.
+	_, err = op.DoneWithContext(ctx, c.ServersClient.Client)
+	return err
+}
+
+// UpdateServer updates a MySQL Server.
+func (c *MySQLServerClient) UpdateServer(ctx context.Context, cr *azuredbv1alpha3.MySQLServer) error {
+	// TODO(muvaf): password update via Update call is supported by Azure but
+	// we don't support that.
+	s := cr.Spec.ForProvider
+	properties := &mysql.ServerUpdateParametersProperties{
+		Version:        mysql.ServerVersion(s.Version),
+		SslEnforcement: mysql.SslEnforcementEnum(s.SSLEnforcement),
+		//ReplicationRole: s.Spec.ForProvider.ReplicationRole,
```

Comment:
Should this be commented out?

Reply:
Yes, `ReplicationRole` is not used in `DefaultCreateMode` which is the only mode we currently support. But since `ServerUpdateParametersProperties` struct is used for all modes, it has that field.

Reply:
```suggestion
		// ReplicationRole is not used in DefeaultCreateMode.
		//ReplicationRole: s.Spec.ForProvider.ReplicationRole,
```